### ISSUE TITLE
2.13.2 breaks *array* of enums

### ISSUE THREAD

**Issue by ThomasLandauer** [doctrine/doctrine2#10063]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 2.13.2

#### Summary

When having an *array* of enums like this:

```php
#[ORM\Column(type: Types::SIMPLE_ARRAY, length: 255, nullable: true, enumType: Color::class)]
private array $colors = [];
```
I'm getting this error:

> "Green,Red,Blue" is not a valid backing value for enum "App\Enum\Color"

I'm guessing that https://github.com/doctrine/orm/pull/10041 broke it, since @michnovka was referring to a *string* column:

```php
#[Column(type: Types::STRING, nullable: false, enumType: OrderStatus::class, options: ['charset' => 'utf8', 'collation' => 'utf8_unicode_ci'])]
private OrderStatus $orderStatus;
```
#### Workaround

For anybody else observing this, a workaround for now is to add this to your `composer.json` to continue using 2.13.1:
```json
"conflict": {
    "doctrine/orm": "2.13.2"
},
```

#### Next Step

Well, I think it's about time to properly document arrays of enums at https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/basic-mapping.html - I'll come up with a PR after getting some feedback here.

**Comment by greg0ire** [comment]
Can you provide a test case for this? Also, please note that the code has already changed since 2.13.2: #10058

**Comment by michnovka** [comment]
@ThomasLandauer I believe that https://github.com/doctrine/orm/pull/10058 fixes this issue, but please do provide a test case, we should push it for CI sake

**Comment by ThomasLandauer** [comment]
There you go - my first attempt for a test here ;-) https://github.com/doctrine/orm/pull/10111
Please take a look, I probably need some help there.

Anyway, the issue itself does indeed look fixed in v2.13.3, so I'm closing here.
